File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/GameRolePlayShowActorMessage.scala





















// Generated on 05/03/2013 10:34:42
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameRolePlayShowActorMessage extends MessageDeserializer {
  val messageId = 5632

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameRolePlayShowActorMessage.messageId

    val informations: GameRolePlayActorInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeUShort(informations.typeId)
      informations.serialize(buf)
    }

  }

  protected def create(informations: GameRolePlayActorInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val informations = types.get[GameRolePlayActorInformations.TypeImpl](buf.readShort());

    create(informations)
  }

}

object GameRolePlayShowActorMessage extends GameRolePlayShowActorMessage {
  case class MessageImpl(informations: GameRolePlayActorInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(informations: GameRolePlayActorInformations.TypeImpl) = MessageImpl(informations)

  def apply(informations: GameRolePlayActorInformations.TypeImpl) = create(informations)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameRolePlayShowActorMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameRolePlayShowActorMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5632;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public GameRolePlayActorInformations informations;
    
    public GameRolePlayShowActorMessage() { }
    
    public GameRolePlayShowActorMessage(GameRolePlayActorInformations informations) {
        this.informations = informations;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(informations.getTypeId());
        informations.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        informations = ProtocolTypeManager.getInstance().build(buf.readShort());
        informations.deserialize(buf);
    }
    
}
